src/Utilities.cpp

```diff
@@ -0,0 +1,16 @@
+#include "Utilities.h"
+
+
+namespace EERAModel
+{
+namespace Utilities
+{
+    std::string toUpper(std::string str)
+    {
+        std::string _temp(str);
```

Comment:
`str` is a local copy of the string passed in to the function. You don't need to make another copy of it. Just transform it directly.